### PR TITLE
enhancement #510: refactored DisciplineTeacher Entity

### LINKED ISSUE #510: Refactor DisciplineTeacher entity
Your task will be to refactor all files, related to **DisciplineTeacher** entity, such as controller, service, responses, DTOs,  etc.

Your steps can be as following:
1. Check service and repository class for potential issues, misspeling or incorect code style.
2. You need to run unit tests for this service and try to enhance them, also check that all necessary service methods are covered.
3. Check responses and DTOs. As we have a lot of changes some methods can return and take undocumentated information.
4. After all responses and DTOs are checked you can separate documentation of endpoints in controller as made in [this](https://github.com/fictadvisor/fictadvisor/pull/497) pr.

### PR DESCRIPTION
**Pull request by bohdanbulakh**
- separated documentation from implementation in DisciplineTeacher controller
- refactored all DTOs and responses, which are related to disciplineTeacher controller
- improved names of mappers methods, related to disciplineTeacher entity

closes #510

### CODE REVIEW

**Review by sifes — APPROVED** [review]

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 1]
```diff
@@ -201,213 +100,108 @@ export class DisciplineTeacherController {
...
-    description: `
-      InvalidBodyException: Question id can not be empty\n
-      InvalidBodyException: Value can not be empty\n
-      InvalidBodyException: User id can not be empty\n
-      InvalidEntityIdException: disciplineTeacher with such id is not found
-    `,
+  @ApiEndpoint({
+    summary: 'Send question`s response disciplineTeacherId',
+    documentation: DisciplineTeacherDocumentation.SEND_RESPONSE,
+    // guards: TelegramGuard,
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:106]: Remove comment

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 2]
```diff
@@ -201,213 +100,108 @@ export class DisciplineTeacherController {
...
   })
-  @UseGuards(TelegramGuard)
   @Post('/:disciplineTeacherId/responses')
-  sendResponse (
+  async sendResponse (
     @Param('disciplineTeacherId', DisciplineTeacherByIdPipe) disciplineTeacherId: string,
+    @Body('userId', UserByIdPipe) userId: string,
+    @Body('questionId', QuestionByIdPipe) questionId: string,
     @Body() body: ResponseDTO
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:113]: Missing trailing comma

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 3]
```diff
@@ -474,46 +232,9 @@ export class DisciplineTeacherController {
...
-    name: 'disciplineTeacherId',
-    required: true,
-    description: 'Discipline teacher id',
-  })
   @ApiEndpoint({
     summary: 'Delete question answer with TEXT type (comment)',
+    documentation: DisciplineTeacherDocumentation.DELETE_COMMENT,
     permissions: PERMISSION.COMMENTS_DELETE,
   })
   @Delete('/:disciplineTeacherId/comments')
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:240]: Unused body object on line 243

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/create.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { DisciplineTeacherExtendedResponse } from '@fictadvisor/utils';
+import { DefaultForbiddenResponse, DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationCreate: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {
+    type: DisciplineTeacherExtendedResponse,
+  },
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/create.ts:11]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/deleteById.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { ApiDocumentationParams } from '../decorators';
+import { DefaultForbiddenResponse, DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationDeleteById: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {},
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/deleteById.ts:8]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/deleteByTeacherAndDiscipline.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { ApiDocumentationParams } from '../decorators';
+import { DefaultForbiddenResponse, DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationDeleteByTeacherAndDiscipline: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {},
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/deleteByTeacherAndDiscipline.ts:8]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/removeDisciplineTeacherFromPoll.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { ApiDocumentationParams } from '../decorators';
+import { DefaultForbiddenResponse, DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationRemoveDisciplineTeacherFromPoll: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {},
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/removeDisciplineTeacherFromPoll.ts:8]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/removeDisciplineTeacherFromPoll.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    InvalidEntityIdException:
+      DisciplineTeacher with such id is not found
+
+      NotSelectedDisciplineException:
+        Current discipline is not selected by this student`,
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/removeDisciplineTeacherFromPoll.ts:13]: Too large indentation

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/sendResponse.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import { QuestionAnswerResponse } from '@fictadvisor/utils';
+import { DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationSendResponse: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {
+    type: QuestionAnswerResponse,
+  },
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/sendResponse.ts:11]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/updateById.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import { DisciplineTeacherExtendedResponse } from '@fictadvisor/utils';
+import { DefaultForbiddenResponse, DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationUpdateById: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {
+    type: DisciplineTeacherExtendedResponse,
+  },
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/updateById.ts:11]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/updateByTeacherAndDiscipline.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+import { DisciplineTeacherExtendedResponse } from '@fictadvisor/utils';
+import { DefaultForbiddenResponse, DefaultUnauthorizedResponse } from '../defaultResponses';
+
+export const DisciplineTeacherDocumentationUpdateByTeacherAndDiscipline: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {
+    type: DisciplineTeacherExtendedResponse,
+  },
+  badRequest: {
+    description: `
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/updateByTeacherAndDiscipline.ts:11]: Add \n

File: fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/sendAnswersByTelegram.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  ok: {},
+  badRequest: {
+    description: `\n
+    InvalidBodyException: 
+      Question id must be a string
+      Question id cannot be empty
+      Value must be a string
+      Value cannot be empty
+      User id must be a string
+      UserId cannot be empty
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/disciplineTeacher/sendAnswersByTelegram.ts:15]: It's better to write 'User id'

File: utils/src/requests/CreateAnswersDTO.ts [hunk 13]
```diff
@@ -27,6 +33,7 @@ export class CreateAnswersWithUserIdDTO extends CreateAnswersDTO {
   @ApiProperty({
     description: 'Id of the user who provided answers',
   })
+  @IsString(validationOptionsMsg('User id must be a string'))
   @IsNotEmpty(validationOptionsMsg('UserId cannot be empty'))
```
Comment (VlO-OlV) [inline utils/src/requests/CreateAnswersDTO.ts:37]: It's better to write 'User id'

File: utils/src/responses/DisciplineResponse.ts
Comment (VlO-OlV) [inline utils/src/responses/DisciplineResponse.ts:1]: Add desciptions

File: utils/src/responses/DisciplineTeacherExtendedResponse.ts
Comment (VlO-OlV) [inline utils/src/responses/DisciplineTeacherExtendedResponse.ts:1]: Add descriptions

File: utils/src/responses/QuestionAnswerResponse.ts
Comment (VlO-OlV) [inline utils/src/responses/QuestionAnswerResponse.ts:1]: Add descriptions

File: fictadvisor-api/src/v2/database/entities/DbQuestionAnswer.ts [hunk 14]
```diff
@@ -1,15 +1,16 @@
...
   questionId: string;
   userId: string;
   value: string;
-  question?: Question;
+  question?: DbQuestion;
   disciplineTeacher?: DisciplineTeacher & {
     discipline?: Discipline & {
       subject: Subject,
     },
     teacher?: Teacher
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/database/entities/DbQuestionAnswer.ts:14]: Missing trailing comma

**Review by VlO-OlV — APPROVED** [review]

**Pull request merged**